### PR TITLE
CI: debug HPU flow

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Checking why the HPU is not running

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @carmocca @akihironitta @borda @tchaton @rohitgr7 @jerome-habana @kaushikb11

### CODE REVIEW

**Comment by Borda** [comment]
the problem here seems to be that we install the package, but even just after it is not found...
```
Installing collected packages: pytorch-lightning
  Running setup.py develop for pytorch-lightning
Successfully installed pytorch-lightning
```
but `pip list` does not list it... :confused:

**Review by jerome-habana — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]

File: .azure/hpu-tests.yml [hunk 1]
```diff
@@ -20,23 +20,44 @@ jobs:
...
         hwinfo --short
+        python --version
+        sudo pip install pip -U
       displayName: 'Instance HW info'
 
     - bash: |
-        pip install -e .[extra] -r requirements/pytorch/test.txt
+        set -e
+        pip --version
+        sudo pip uninstall -y lightning pytorch-lightning
```
Comment (akihironitta) [inline .azure/hpu-tests.yml:46]: `sudo` here is required to avoid `ModuleNotFoundError` or something, right?

Comment (akihironitta) [reply .azure/hpu-tests.yml:46]: Never mind, I just realised that it's `uninstall` but not `install`.

Comment (Borda) [reply .azure/hpu-tests.yml:46]: sudo was for install hwinfo

File: .azure/hpu-tests.yml [hunk 2]
```diff
@@ -20,23 +20,44 @@ jobs:
     timeoutInMinutes: "10"
     # how much time to give 'run always even if cancelled tasks' before stopping them
     cancelTimeoutInMinutes: "2"
-    pool: intel-hpus
+    pool: habana-gaudi-hpus
```
Comment (kaushikb11) [inline .azure/hpu-tests.yml:23]: nit: `hpus` has Habana term in it

Comment (Borda) [reply .azure/hpu-tests.yml:23]: well, yes, but I still wanted to preserve it as all others have it...
I there are temporary two pools so we can rename it later :rabbit:

File: dockers/ci-runner-hpu/start.sh [hunk 3]
```diff
@@ -93,4 +93,4 @@ trap 'cleanup; exit 143' TERM
 
 # To be aware of TERM and INT signals call run.sh
 # Running it with the --once flag at the end will shut down the agent after the build is executed
-./run.sh --once & wait $!
+./run.sh & wait $!
```
Comment (carmocca) [inline dockers/ci-runner-hpu/start.sh:96]: How is this different from running the script?

Also the comment above is outdated

**Review by carmocca — APPROVED** [review]

File: dockers/ci-runner-hpu/start.sh (hunk 3 above)
Comment (carmocca) [reply dockers/ci-runner-hpu/start.sh:96]: Will you do the change in a different PR?

**Pull request merged**